File: empl/php/Technicien.php
<?php
class Technicien extends Employe
{
    private string $grade;

    public function __construct($nom, $age, $salaire, $NumSecu, $grade)
    {
        parent::__construct($nom, $age, $salaire, $NumSecu);
        $this->grade = $grade;
    }

    public function __get($name): mixed
    {
        if ($name === 'grade') {
            return $this->grade;
        }
        return parent::__get($name);
    }

    public function __set($name, $value): void
    {
        if ($name === 'grade') {
            $this->grade = $value;
        } else {
            parent::__set($name, $value);
        }
    }

    public function prime(): int
    {
        return match ($this->grade) {
            'A' => 300,
            'B' => 200,
            'C' => 100,
            default => 0,
        };
    }

    public function calculer_salaire(): float
    {
        $salaire_base = parent::calculer_salaire();
        return $salaire_base + $this->prime();
    }

    public function afficher(): string
    {
        $info_parent = parent::afficher();
        return "{$info_parent}, Grade: {$this->grade}, prime: {$this->prime()}, Salaire Total: {$this->calculer_salaire()}";
    }
}

File: empl/Python/Technicien.py
from Employe import Employe


class Technicien(Employe):
    def __init__(self, nom, age, salaire, numero_securite_sociale, grade):
        super().__init__(nom, age, salaire, numero_securite_sociale)
        self.grade = grade

    def prime(self):
        if self.grade == "A":
            return 300
        elif self.grade == "B":
            return 200
        elif self.grade == "C":
            return 100
        else:
            return 0

    def calculer_salaire(self):
        return super().calculer_salaire() + self.prime()

    def __repr__(self):
        return (
            super().__repr__()
            + f", Grade: {self.grade}, Prime: {self.prime()}, Salaire avec prime: {self.calculer_salaire()}"
        )
